### PR TITLE
Update chef-workstation to include URLs for Intel and ARM packages

### PR DESCRIPTION
**Pull request by vkarve-chef**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making all changes to a cask, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [ ] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [ ] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [ ] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [ ] `brew audit --new-cask <cask>` worked successfully.
- [ ] `brew install --cask <cask>` worked successfully.
- [ ] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

File: Casks/chef-workstation.rb [hunk 1]
```diff
@@ -1,20 +1,31 @@
...
+  sha256_intel = "b957dca3a7be28373a718e95f88290c5be928985ac08b37b8c83b1ac1330d4db"
+  sha256_arm = "728ff9458a7f73618a9a34ccd5cb41962ce061f2d67ef2e58e19dd0251c528a5"
+
+  version "22.1.778"
+
+  if Hardware::CPU.intel?
+    sha256 sha256_intel.to_s
+  else
+    sha256 sha256_arm.to_s
+  end
```
Comment (miccal) [inline Casks/chef-workstation.rb:14]:
```suggestion
  version "22.1.778"

  if Hardware::CPU.intel?
    sha256 "b957dca3a7be28373a718e95f88290c5be928985ac08b37b8c83b1ac1330d4db"
  else
    sha256 "728ff9458a7f73618a9a34ccd5cb41962ce061f2d67ef2e58e19dd0251c528a5"
  end
```

**Review by miccal — CHANGES_REQUESTED** [review]

File: Casks/chef-workstation.rb (hunk 1 above)
Comment (vkarve-chef) [reply Casks/chef-workstation.rb:14]: @miccal, an earlier version did have the sha256 as specified in the suggested change. i intentionally moved it outside the if-else block. we have a script that updates the package version, sha256 values and then creates a PR. writing it this way makes it easier for find-replace using regex.
if bump-cask-pr automation handles the condition based sha256 update, please let me know. i can then update cask as you suggest.

Comment (miccal) [reply Casks/chef-workstation.rb:14]: Yes `bump-cask-pr` handles this fine -- please change it to be consistent with our other Casks.

Comment (vkarve-chef) [reply Casks/chef-workstation.rb]: done

Comment (vkarve-chef) [reply Casks/chef-workstation.rb]: done

File: Casks/chef-workstation.rb [hunk 2]
```diff
@@ -1,20 +1,28 @@
 cask "chef-workstation" do
-  version "22.1.745"
-  sha256 "f6beaa7b00ecb062d541648e6562c9dba3c5380176c2cef612518b114cee712a"
+  arch = Hardware::CPU.intel? ? "x86_64" : "arm64"
+  macos_version = Hardware::CPU.intel? ? "10.15" : "11"
```
Comment (carlocab) [inline Casks/chef-workstation.rb:3]:
```suggestion
  arch, macos_version = Hardware::CPU.intel? ? ["x86_64", "10.15"] : ["arm64", "11"]
```

**Comment by vkarve-chef** [comment]
closing as the same changes are added in another [PR](https://github.com/Homebrew/homebrew-cask/pull/118816) that addresses the merge conflict.

**Pull request closed without merging**